src/main/java/org/openrewrite/java/testing/junit5/AddParameterizedTestAnnotation.java

```diff
@@ -0,0 +1,78 @@
+package org.openrewrite.java.testing.junit5;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.java.AnnotationMatcher;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.tree.*;
+
+import java.util.List;
+
+public class AddParameterizedTestAnnotation extends Recipe {
+    private static final AnnotationMatcher TEST_ANNOTATION_MATCHER = new AnnotationMatcher("@org.junit.jupiter.api.Test");
+
+    @Override
+    public String getDisplayName() {
+        return "Add missing @ParameterizedTest annotation when @ValueSource is used or " +
+               "replace @Test with @ParameterizedTest";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Add missing @ParameterizedTest annotation when @ValueSource is used or " +
+               "replace @Test with @ParameterizedTest.";
+    }
+
+    @Override
+    public JavaIsoVisitor<ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            // checks if annotations matches ValueSource, CsvSource or MethodSource
+            public boolean checkForValueAnnotations(J.Annotation ann) {
+                boolean valSourceCheck = (ann.getSimpleName().equals("ValueSource"));
+                boolean csvSourceCheck = (ann.getSimpleName().equals("CsvSource"));
+                boolean methodSourceCheck = (ann.getSimpleName().equals("MethodSource"));
```

Comment:
You're missing out on matching quite a few other `@...Source` annotations still; see this section of the docs for an overview: https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests